ca/certificates.go

```diff
@@ -579,6 +562,79 @@ func ValidateCertChain(rootPool *x509.CertPool, certs []byte, allowExpired bool)
 	return parsedCerts, nil
 }
 
+// newLocalSigner validates the signing cert and signing key to create a local signer, which accepts a crypto signer and a cert
+func newLocalSigner(keyBytes, certBytes []byte, certExpiry time.Duration, rootPool, intermediatePool *x509.CertPool) (*LocalSigner, error) {
+	switch {
+	case len(keyBytes) == 0 && len(certBytes) == 0: // not valid signer, but no error
+		return nil, nil
+	case len(keyBytes) > 0 && len(certBytes) > 0: // possibly valid signer
+	default:
+		return nil, errors.New("must provide both a signing key and a signing cert, or neither")
+	}
```

Comment:
I'd suggest moving this check out of the call to `newLocalSigner`, so we can avoid the `return nil, nil` case. By convention, a function that returns a pointer and an error will generally return a valid pointer if the error is `nil`. Violating this isn't the end of the world, but changing `newLocalSigner` to a function that we only call when we expect it to return a `LocalSigner` or an error would let us stick to this convention.